tofu/tests/tests01_geom/test_04_sampling.py

```diff
@@ -301,23 +326,29 @@ def test25_sa_integ_map(ves_poly=VPoly, debug=1):
         # import pdb; pdb.set_trace()
         ax.scatter(pts[0, :], pts[1, :],
                    marker="s", edgecolors="None",
-                   s=10, c=sa_map.flatten(),
-                   vmin=0, vmax=max(sa_map.max(), sa_map_py.max()))
+                   s=10, c=sa_map_cy.flatten(),
+                   vmin=0, vmax=max(sa_map_cy.max(), sa_map_py.max()))
         ax.set_title("cython function")
         ax = plt.subplot(122)
         ax.scatter(pts[0, :], pts[1, :],
                    marker="s", edgecolors="None",
                    s=10, c=sa_map_py.flatten(),
-                   vmin=0, vmax=max(sa_map.max(), sa_map_py.max()))
+                   vmin=0, vmax=max(sa_map_cy.max(), sa_map_py.max()))
         ax.set_title("python reconstruction")
         plt.savefig("comparaison")
 
-    print("max error approx =", np.max(np.abs(sa_map_py - sa_map)/sa_map_py))
-    print("max error exacts =", np.max(np.abs(sa_map_py_ex - sa_map)/sa_map_py_ex))
-    print("max error python =", np.max(np.abs(sa_map_py - sa_map_py_ex)/sa_map_py_ex))
-
-    assert np.allclose(sa_map, sa_map_py, rtol=1)
-    assert np.allclose(sa_map, sa_map_py_ex, rtol=1)
+    print("max error approx py vs cy =",
+          np.max(np.abs(sa_map_py - sa_map_cy) / sa_map_py))
+    print("max error exacts py vs cy =",
+          np.max(np.abs(sa_map_py_ex - sa_map_cy_ex) / sa_map_py_ex))
+    print("max error python approx vs exact =",
+          np.max(np.abs(sa_map_py - sa_map_py_ex) / sa_map_py_ex))
+    print("max error cython approx vs exact =",
+          np.max(np.abs(sa_map_cy - sa_map_cy_ex) / sa_map_cy_ex))
+
+    assert np.allclose(sa_map_cy, sa_map_py, rtol=1)
+    assert np.allclose(sa_map_cy_ex, sa_map_py_ex, rtol=1)
+    assert np.allclose(sa_map_cy, sa_map_cy_ex, rtol=1)
```

Comment:
Nice, good that it is working and passing tests,
I guess now we are going to decrease the relative tolerance to right ? and set atol = 0 too ?
Any idea what was wrong ?

Reply:
Last time we spoke it was working. I just added a few things that were on my TODO list.

> decrease the relative tolerance to right

What do you mean by this ?

Reply:
> What do you mean by this ?

Sorry, I meant decrease the relative tolerance to 0.
Now calling `np.allclose(rtol=1) `means we accept 100 % of relatve error + absolute error
We need to call something like `np.allclose(atol=0, rtol=1e-14, equal_nan=True)` or something close (1e-12 or even 1e-10 would be acceptable I guess)
See here for [doc](https://numpy.org/doc/stable/reference/generated/numpy.allclose.html).